backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/floatingnotification/internal/BpkFloatingNotification.kt

```diff
@@ -0,0 +1,144 @@
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import net.skyscanner.backpack.compose.button.BpkButton
+import net.skyscanner.backpack.compose.button.BpkButtonSize
+import net.skyscanner.backpack.compose.button.BpkButtonType
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.icon.BpkIconSize
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkBorderRadius
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+
+@Composable
+internal fun BpkFloatingNotificationImpl(
+  data: BpkFloatingNotificationData,
+  modifier: Modifier = Modifier,
+) {
+
+  Snackbar(
+    modifier = modifier
+      .floatingNotificationSize(LocalConfiguration.current)
+      .padding(start = BpkSpacing.Base, end = BpkSpacing.Base),
+    shape = RoundedCornerShape(BpkBorderRadius.Md),
+    backgroundColor = BpkTheme.colors.corePrimary,
+    contentColor = BpkTheme.colors.textOnDark,
+  ) {
+
+    Row(
+      modifier = Modifier.fillMaxHeight(),
+      horizontalArrangement = Arrangement.spacedBy(BpkSpacing.Base),
+      verticalAlignment = Alignment.CenterVertically,
+    ) {
+
+      data.icon?.let { icon ->
+        BpkIcon(
+          icon = icon,
+          contentDescription = null,
+          size = BpkIconSize.Small,
+        )
+      }
+
+      BpkText(
+        modifier = Modifier.weight(1f),
+        text = data.text,
+        maxLines = 2,
+        style = BpkTheme.typography.footnote,
+        overflow = TextOverflow.Ellipsis,
+      )
+
+      data.cta?.let { cta ->
+        BpkButton(
+          text = cta,
+          type = BpkButtonType.LinkOnDark,
+          onClick = { data.performAction() },
+          size = BpkButtonSize.Default,
+        )
+      }
+    }
+  }
+}
+
+@OptIn(ExperimentalAnimationApi::class)
+internal fun floatingNotificationTransforms(): AnimatedContentScope<BpkFloatingNotificationData?>.() -> ContentTransform =
+  {
+    ContentTransform(
+      targetContentEnter = fadeIn(tween(TRANSITION_DURATION)) + slideInVertically(tween(TRANSITION_DURATION)) { it / 2 },
+      initialContentExit = fadeOut(tween(TRANSITION_DURATION)) + slideOutVertically(tween(TRANSITION_DURATION)) { it / 2 },
+    )
+  }
+
+private object BpkFloatingNotificationSizes {
+  const val SMALL_MOBILE_MAX_WIDTH = 360
+  const val MOBILE_MAX_WIDTH = 512
+
+  val SMALL_MOBILE_REQUIRED_WIDTH = 288.dp
+  val SMALL_MOBILE_REQUIRED_HEIGHT = 52.dp
+  val MOBILE_REQUIRED_WIDTH = 312.dp
+  val MOBILE_REQUIRED_HEIGHT = 52.dp
+  val TABLET_REQUIRED_WIDTH = 400.dp
+  val TABLET_REQUIRED_HEIGHT = 72.dp
+}
+
+private fun Modifier.floatingNotificationSize(configuration: Configuration): Modifier =
+  run {
+    if (configuration.screenWidthDp < BpkFloatingNotificationSizes.SMALL_MOBILE_MAX_WIDTH) {
+      requiredSize(
```

Comment:
You can store width/heigt pair as DpSize, e.g. val SizeSmall = DpSize(288.dp, 38.dp) – it'll make it easier to read